Remove dummy dataset writing from RecordComponent::flush()

franzpoeschel opened the pull request:
See #489 and #512 for a description. The exactly same thing happens also in [`RecordComponent.cpp`](https://github.com/openPMD/openPMD-api/blob/dev/src/RecordComponent.cpp#L116). PR #512 did not catch this.

franzpoeschel committed:
Remove dummy dataset writing from RecordComponent::flush()

ax3l commented:
CI seams unhappy.
You might want to implement the default for such records and attributes on `Series::close()` or similar.

franzpoeschel commented:
The problem is with the `fileBased_write_empty_test` which triggers exactly the situation that I have described [here](https://github.com/openPMD/openPMD-api/pull/512#issue-278721748):
> It is a design decision in ADIOS (also ADIOS2) not to physically create variables (used for openPMD datasets) unless data is actually written and openPMD can leverage this behavior. The workaround removed in this PR tries to circumvent this behavior. _Needs to be documented in this case_.

Relevant part of the test code:
```cpp
    Dataset dset = Dataset(Datatype::DOUBLE, {2});
    {
        Series o = Series("../samples/subdir/serial_fileBased_write%T." + backend, AccessType::CREATE);

        ParticleSpecies& e_1 = o.iterations[1].particles["e"];
        e_1["position"][RecordComponent::SCALAR].resetDataset(dset);
        e_1["positionOffset"][RecordComponent::SCALAR].resetDataset(dset);
        o.iterations[1].setTime(1.f);
        ParticleSpecies& e_2 = o.iterations[2].particles["e"];
        e_2["position"][RecordComponent::SCALAR].resetDataset(dset);
        e_2["positionOffset"][RecordComponent::SCALAR].resetDataset(dset);
        o.iterations[2].setTime(2.f);
        ParticleSpecies& e_3 = o.iterations[3].particles["e"];
        e_3["position"][RecordComponent::SCALAR].resetDataset(dset);
        e_3["positionOffset"][RecordComponent::SCALAR].resetDataset(dset);
        o.iterations[3].setTime(3.f);
    }
```
Since no data is ever written, the ADIOS variable is not created in storage and not read back in.

Implementing a default as you suggest would be equivalent to writing dummy data, just at a later point.
If we go for this, we should keep #516 in mind. Should we?

franzpoeschel committed:
Remove fileBased_write_empty test

This test tested exactly the behaviour that this branch removes.
Also remove a number of duplicate tests.
